src/adapters/http/index.ts

```diff
@@ -0,0 +1,108 @@
+import Adapter from '../../lib/adapter.js';
+import GleeMessage from '../../lib/message.js';
+import http from 'http';
+import { validateData } from '../../lib/util.js';
+import GleeError from '../../errors/glee-error.js';
+import * as url from 'url';
+
+class HttpAdapter extends Adapter {
+    private res: any[] = []
+
+    name(): string {
+        return 'HTTP server'
+    }
+
+    async connect(): Promise<this> {
+        return this._connect();
+    }
+
+    async send(message: GleeMessage): Promise<void> {
+        return this._send(message);
+    }
+
+    _connect(): Promise<this> {
+        return new Promise((resolve, reject) => {
+            const serverUrl = new URL(this.serverUrlExpanded);
+            const httpServer = this.glee.options?.http?.server || http.createServer();
```

Comment:
This would change in the future, as per https://github.com/asyncapi/glee/pull/351 so do give it a look and also comment if you need any specific config for HTTP adapter be it authentication.

Reply:
@ritik307 As I have told you earlier, this has changed, now there is a function that will resolve and give you the required config object. 
You need to do something like this 


```suggestion
            const httpServer = await this.resolveProtocolConfig('ws')
```